### ISSUE TITLE
Getting strange uncatchable errors with users that block bot

### ISSUE THREAD

**Issue by Faydaen** [irazasyed/telegram-bot-sdk#169]
Hello i'm getting uncatchable error while trying to sendMessage to user who had blocked bot, api says that message was sent successfully but after that i get an error in console
`PHP Fatal error:  Uncaught GuzzleHttp\Exception\ClientException: Client error:'POST https://api.telegram.org/bot{my_token}/sendMessage' resulted in a '403 Forbidden' response:
{"ok":false,"error_code":403,"description":"[Error : 403 : Bot was blocked by the user]"}
 in /home/ghost/web/easeFinance/bots/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:107
Stack trace:
#0 /home/ghost/web/easeFinance/bots/vendor/guzzlehttp/guzzle/src/Middleware.php(65): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response))
#1 /home/ghost/web/easeFinance/bots/vendor/guzzlehttp/promises/src/Promise.php(201): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /home/ghost/web/easeFinance/bots/vendor/guzzlehttp/promises/src/Promise.php(154): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), Array)
#3 /home/ghost/web/easeFinance/bots/vendor/guzzlehttp/promises/src/TaskQueue.php(61): GuzzleHttp in /home/ghost/web/easeFinance/bots/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 107`

tried to debug sendRequest method, there i get $response as a Promise class with no body and errors in it, but i think this happens after API get's telegram response.

**Comment by irazasyed** [comment]
If the bot is blocked, you can't send them any message. It's simple. There is no way to bypass that.

As far as the response is concerned, If the request is async, then there is no response but if it's not, then you'll get the response (Properly mapped to the appropriate object).

**Comment by elvaluta** [comment]
Is there any way to catch this exception and to log it or to continue execute the script? I have 2 different bots, and one of them is falling down bcs of this error. Another one continue to elaborate requests, and I don't know how to solve it. Bots have same configuration, turning on/off assync requests doesn't help.
Thank you.

**Comment by firegore2** [comment]
I have the same problem... i don't know how to catch the error and process it to avoid the break script

**Comment by irazasyed** [comment]
@firegore2 Please create a new issue with more details and your code.

**Comment by devharandi** [comment]
for by pass looping in async(true)

in GuzzleHttpClient file  in > vendor/irazasyed/telegram-bot-sdk/src/HttpClients

change function

wtih
    /**
     * Unwrap Promises.
     */
    public function __destruct()
    {
        try
        {
            Promise\unwrap(self::$promises);
        }
        catch(\Exception $e)
        {
            \Log::error($e->getMessage());
        }
    }

**Comment by foremtehan** [comment]
How can i catch asynchronous requests? Even i wrap my codes in try/catch it throws a error, @devharandi is there another way to not mainpulate the vendor codes?

**Comment by jonnywilliamson** [comment]
If the try catch is working you may be trying to catch the WRONG type of exception.

Please post a copy of the exact error message/exception you saw

**Comment by foremtehan** [comment]
@jonnywilliamson 

```php

try {
   $tg =  (new Api($apiKey))->setAsyncRequest(true);
   $tg->sendMessage($params);
} catch (TelegramSDKException $e) {
    echo $e->getMessage();
}

```
